test/infrastructure/docker/docker/util.go

```diff
@@ -17,29 +17,36 @@ limitations under the License.
 package docker
 
 import (
+	"context"
 	"fmt"
 	"strings"
 
+	dockertypes "github.com/docker/docker/api/types"
+	dockerfilters "github.com/docker/docker/api/types/filters"
 	"github.com/pkg/errors"
 	"sigs.k8s.io/cluster-api/test/infrastructure/docker/docker/types"
-	"sigs.k8s.io/kind/pkg/exec"
 )
 
 const clusterLabelKey = "io.x-k8s.kind.cluster"
 const nodeRoleLabelKey = "io.x-k8s.kind.role"
+const filterLabel = "label"
+const filterName = "name"
 
-// clusterLabel returns the label applied to all the containers in a cluster.
-func clusterLabel(name string) string {
-	return toLabel(clusterLabelKey, name)
-}
+type Filter map[string]map[string]string
+
+// AddKeyValue adds a filter with a single name.
+func (f Filter) AddKeyValue(key, value string) {
```

Comment:
What about
```suggestion
func (f FilterBuilder) AddFilterWithValue(filter, value string) {
```
I'm trying to get away from using key, which confused me at the first read...

Reply:
Done